### PR TITLE
Added smpp

### PR DESCRIPTION
**Pull request by akoidan**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -236,6 +236,7 @@ dependencies {
     compile "com.icthh.xm.commons:xm-commons-lep:${xm_commons_version}"
     compile("com.icthh.xm.commons:xm-commons-ms-web:${xm_commons_version}") 
     compile("com.icthh.xm.commons:xm-commons-permission:${xm_commons_version}")
+    compile 'org.jsmpp:jsmpp:2.3.7'
```
Comment (vkirichenko) [inline build.gradle]: Put version to build.properties and move dependency before xm dependensies

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            log.info("Message submitted, message_id is {}", messageId);
+        } catch (PDUException e) {
+            log.error("Invalid PDU parameter {}", e);
+        } catch (ResponseTimeoutException e) {
+            log.error("Response timeout {}", e);
+        } catch (InvalidResponseException e) {
+            log.error("Received invalid response {}", e);
+        } catch (NegativeResponseException e) {
+            log.error("Receive negative response {}", e);
+        } catch (IOException e) {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: hmm, when sms was not sended client received 200 Ok ?

File: src/main/docker/smpp/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.8
```
Comment (ishkurko) [inline src/main/docker/smpp/Dockerfile]: Maybe folder rename into `smpp-emulator`?

File: src/main/docker/smpp/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+RUN cd /home &&\
+apk --update add openjdk7-jre &&\
+wget -O /home/file.zip https://drive.google.com/uc?id=0B7YOzDQMt_TmaWN4UkJFa3M0Rkk &&\
```
Comment (ishkurko) [inline src/main/docker/smpp/Dockerfile]: Why download for the google drive and not from the official site?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void send(String destAdrrs, String message) {
+        SMPPSession session = null;
+        try {
+            session = this.smppConfiguration.getSession();
+            String messageId = session.submitShortMessage(
+                "CMT",
+                TypeOfNumber.INTERNATIONAL,
+                NumberingPlanIndicator.UNKNOWN,
+                "1616",
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: magic number?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                TypeOfNumber.INTERNATIONAL,
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: replace by static import

File: src/main/java/com/icthh/xm/tmf/ms/communication/config/SmppConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public SMPPSession getSession() throws IOException {
+        SMPPSession session = new SMPPSession();
+        ApplicationProperties.Smpp smpp = appProps.getSmpp();
+        BindParameter bindParam = new BindParameter(
+            BindType.BIND_TX,
+            smpp.getSystemId(),
+            smpp.getPassword(),
+            "cp",
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/config/SmppConfiguration.java]: What is `cp`?

File: src/main/java/com/icthh/xm/tmf/ms/communication/config/SmppConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    private final ApplicationProperties appProps;
+
+    protected SmppConfiguration(ApplicationProperties appProps) {
+        this.appProps = appProps;
+    }
+
+    public SMPPSession getSession() throws IOException {
+        SMPPSession session = new SMPPSession();
+        ApplicationProperties.Smpp smpp = appProps.getSmpp();
+        BindParameter bindParam = new BindParameter(
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/config/SmppConfiguration.java]: Let move all fields in configuration properties.

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import org.jsmpp.util.AbsoluteTimeFormatter;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Date;
+
+@Slf4j
+@Component
+public class SmppService {
+
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: Why 2 empty lines?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+    private final SmppConfiguration smppConfiguration;
+    private final AbsoluteTimeFormatter timeFormatter;
+
+    public SmppService(SmppConfiguration smppConfiguration) {
+        this.smppConfiguration = smppConfiguration;
+        this.timeFormatter = new AbsoluteTimeFormatter();
+    }
+
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: Why two empty lines?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void send(String destAdrrs, String message) {
+        SMPPSession session = null;
+        try {
+            session = this.smppConfiguration.getSession();
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: Should we always get session and the unbind?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            String messageId = session.submitShortMessage(
+                "CMT",
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: What is CMT? And other hardcoded value? Let's move all of them in configuration.

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    private final SmppConfiguration smppConfiguration;
+    private final AbsoluteTimeFormatter timeFormatter;
+
+    public SmppService(SmppConfiguration smppConfiguration) {
+        this.smppConfiguration = smppConfiguration;
+        this.timeFormatter = new AbsoluteTimeFormatter();
+    }
+
+
+    public void send(String destAdrrs, String message) {
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: Where is a test for this method?

Comment (ishkurko) [reply src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: And why it is not called from the API?

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java (hunk 2 above)
Comment (ishkurko) [reply src/main/java/com/icthh/xm/tmf/ms/communication/service/SmppService.java]: @akoidan Yes. Please, do not hide errors.

**Pull request merged**